crates/ironrdp-server/src/encoder/bitmap.rs

```diff
@@ -20,20 +20,20 @@ impl BitmapEncoder {
     }
 
     pub(crate) fn encode(&mut self, bitmap: &BitmapUpdate, output: &mut [u8]) -> Result<usize, PduError> {
-        let row_len = bitmap.width * u32::from(bitmap.format.bytes_per_pixel());
-        let chunk_height = u32::from(u16::MAX) / row_len;
+        let row_len = bitmap.width.get() as usize * bitmap.format.bytes_per_pixel() as usize;
+        let chunk_height = u16::MAX as usize / row_len;
```

Comment:
Suggestion: `as` conversions should be avoided when possible. In this case it’s possible to use `usize::from` to convert from `u16` in a guaranteed to be lossless way.

Rationale: `as` conversions will perform many kinds of conversions, including silently lossy conversions and dangerous coercions.